gulp-cucumber exits with code 0 although tests are failing

See travis-ci build output here: https://travis-ci.org/joniba/bidding-express-test/jobs/88510330
Source code here: https://github.com/joniba/bidding-express-test

Could it have to do with my BeforeFeatures and AfterFeatures hooks (https://github.com/joniba/bidding-express-test/blob/master/features/support/hooks.js)? I'm trying to start the server before running the BDD tests and shut down the server once they are done. I'm calling process.exit because otherwise running cucumber via gulp-cucumber does not exit.

For comparison, the same concept is working when using mocha: https://travis-ci.org/joniba/bidding-express-test/jobs/88523877 (5 tests failed and gulp-mocha exits with code 1, not 0, causing the build to fail).

Hello!

I looked at your code and did some manipulations and everything seems to work fine.

First of all I removed `['cucumber']` section from running cucumber task in tge gulpfile. So, your section for running cucumber tests looks like this:
 ```js
gulp.task('cucumber', function () {
...
});
```
You don't need it because you are running server in hooks.

Also I updated and used newest version of `gulp-cucumber`.

So, now output looks like this
```
⇒ gulp cucumber
[22:53:54] Using gulpfile ~/my/bidding-express-test/gulpfile.js
[22:53:54] Starting 'cucumber'...
Feature: Get all bids

  As an administrator
  I want to get all bids made for a particular item
  So that I can see the status of the bidding on that item

  Scenario: Require credentials                                    # features/get-bids.feature:6
    Given I am not authenticated                                   # features/get-bids.feature:7
[TypeError: buf.copy is not a function]
GET /api/bids 401 13.189 ms - -
    When I get the bids for an item                                # features/get-bids.feature:8
    Then I should get an error indicating that I am not authorized # features/get-bids.feature:9

  Scenario: Get all bids                     # features/get-bids.feature:11
    Given I am authenticated                 # features/get-bids.feature:12
    And there are some bids for item '19032' # features/get-bids.feature:13
[TypeError: buf.copy is not a function]
GET /api/bids 401 32.013 ms - -
    When I get the bids for an item          # features/get-bids.feature:14
    Then I should see all associated bids    # features/get-bids.feature:15
      AssertionError: expected 200 to equal 401
          at null.<anonymous> (/Users/vgamula/my/bidding-express-test/features/steps/get-bids.steps.js:34:12)

Failing scenarios:
features/get-bids.feature:11 # Scenario: Get all bids

2 scenarios (1 failed, 1 passed)
7 steps (1 failed, 6 passed)
0m00.106s
---shutting down server---

events.js:141
      throw er; // Unhandled 'error' event
      ^
Error: Cucumber tests failed!
``` 

Error throws and everything is working fine.


You are free to reopen this issue if you would have any questions

--- a/index.js
+++ b/index.js
@@ -69,7 +69,6 @@ module.exports = function(options) {
 
         var stream = this;
         Cucumber.Cli(argv).run(function(succeeded) {
-            stream.emit('end');
             if (succeeded) {
                 callback();
             } else {

